torch/csrc/jit/tensorexpr/eval.cpp

```diff
@@ -985,22 +993,20 @@ void SimpleIREvaluator::call(const std::vector<CallArg>& args) {
   USE_TRIGGER(simple_ir_eval_executed);
 }
 
-void SimpleIREvaluator::call_raw(const std::vector<void*>& args) {
-  throw std::runtime_error(
-      "SimpleIREvaluator::call_raw is not implemented yet");
-}
-
-void SimpleIREvaluator::bindArg(const BufferArg& bufArg, const CallArg& data) {
+void SimpleIREvaluator::bindArg(const BufferArg& bufArg, void* data) {
   if (!bufArg.isVar()) {
-    impl_->bindBuf(bufArg.buf(), data.data());
+    impl_->bindBuf(bufArg.buf(), data);
     return;
   }
 
   switch (bufArg.dtype().scalar_type()) {
-#define TYPE_CASE(Type, Name)                        \
-  case ScalarType::Name:                             \
-    impl_->bindVar(bufArg.var(), *data.Name##Ptr()); \
-    break;
+#define TYPE_CASE(Type, Name)                 \
+  case ScalarType::Name: {                    \
+    Type typed_data;                          \
+    memcpy(&typed_data, &data, sizeof(Type)); \
```

Comment:
We seem to be adding an extra `memcpy` here. Isn't this increasing the overhead? Should we retain the old code for when it is called using `CallArg`?